### PR TITLE
[https://nvbugs/6250439][fix] Correct speculative XQA attention sinks

### PR DESCRIPTION
**Pull request by dongfengy**
## Description

Speculative HMMA XQA flattens output rows as `[query token, head]`, but attention-sink normalization only covered the first head group. For GPT-OSS Eagle, later speculative query-token rows therefore omitted the learned per-head sink contribution and produced corrupted target logits.

This change:

- repeats each per-head attention sink across every speculative query-token row;
- removes a duplicated speculative sliding-window row offset whose units did not match the token sequence length;
- makes the CPU reference attention-sink aware; and
- adds regression coverage for Eagle depths 1 and 3, batch sizes 1 and 4, and TP1/TP4 KV-head shapes.

The change is internal to XQA. It introduces no public API change, dependency, ownership change, documentation requirement, or architecture-diagram change.

## Test Coverage

- Pre-commit checks on all four modified XQA files and `git diff --check`.
- Native SM90 release build with `FAST_BUILD=OFF`.
- Focused HMMA reference regression on H100 PCIe and H200 for depth 1/3, batch 1/4, and 8/2 local KV heads. All cases passed with maximum absolute error `0.003906`.
- Full H200 TP4/EP1 Eagle depth-3 GSM8K: flexible exact match `0.4610`, strict exact match `0.2904`, return code `0`. This clears the NVBug `0.42` threshold.
- Full H100 PCIe TP4/EP1 Eagle depth-3 GSM8K: flexible exact match `0.4685`, strict exact match `0.2866`, return code `0`.

The focused and full Hopper validation will also be rerun on the rebased PR head.

## PR Checklist

Please review the following before submitting your PR:

- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of my knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work)).
- No public API change is introduced, so an API compatibility label is not required.
- No new dependencies are introduced.
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) does not need an update because ownership is unchanged.
- No user-facing documentation or architecture diagram changes are needed for this kernel correctness fix.
- `@jhaotingc` and `@lowsfer` are requested as the primary XQA/speculative-decoding reviewers.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, comment `/bot help`.

### CODE REVIEW

**Review by jhaotingc — APPROVED** [review]
Thank you for the fix!

File: cpp/kernels/xqa/test/test.cpp [hunk 1]
```diff
@@ -1361,6 +1361,20 @@ TEST(RefCheck, llama_V2_70b_3)
...
+    runTest<8, HEAD_GROUP_SIZE, 4>(1, 260, false, true, true, false, true, ~0U, 128);
+    runTest<8, HEAD_GROUP_SIZE, 2>(4, 258, false, true, true, false, true, ~0U, 128);
+    runTest<8, HEAD_GROUP_SIZE, 4>(4, 260, false, true, true, false, true, ~0U, 128);
+    runTest<2, HEAD_GROUP_SIZE, 2>(1, 258, false, true, true, false, true, ~0U, 128);
+    runTest<2, HEAD_GROUP_SIZE, 4>(1, 260, false, true, true, false, true, ~0U, 128);
+    runTest<2, HEAD_GROUP_SIZE, 2>(4, 258, false, true, true, false, true, ~0U, 128);
+    runTest<2, HEAD_GROUP_SIZE, 4>(4, 260, false, true, true, false, true, ~0U, 128);
+}
+#endif
+
```
[the remark this answers is not in the document]
Comment (dongfengy) [reply cpp/kernels/xqa/test/test.cpp:1377]: This test is already enclosed by the outer #if SPEC_DEC; adding it to the nested guard would be a no-op
